internal/server/list_clients_test.go

```diff
@@ -130,3 +142,572 @@ func TestGetFiltersFromParamsWithAllFilters(t *testing.T) {
 	assert.Equal(t, expectedResponseForAccommodation, resultAccommodation)
 	assert.Equal(t, expectedResponseForSupervisionLevel, resultSupervisionLevel)
 }
+
+func TestPostBulkAssuranceVisitTasksReturnsErrorWithDueDate(t *testing.T) {
+	assert := assert.New(t)
+	client := &mockDeputyHubClientInformation{}
+
+	form := url.Values{}
+	form.Add("due-date", "")
+
+	template := &mockTemplates{}
+	w := httptest.NewRecorder()
+	r, _ := http.NewRequest("POST", "/1/bulk-assurance-visit-tasks", strings.NewReader(form.Encode()))
+
+	returnedError := renderTemplateForClientTab(client, template)(AppVars{}, w, r)
+
+	expectedValidationErrors := sirius.ValidationErrors{
+		"due-date": {"": "Enter a due date"},
+	}
+
+	assert.Equal(ListClientsVars{
+		Clients: sirius.ClientList{
+			Clients: []sirius.DeputyClient(nil),
+			Pages: sirius.Page{
+				PageCurrent: 0,
```

Comment:
Golang sets variables to their default value if not specified. This file would be a lot cleaner if any of the variables you set to 0 or "" were just not set in the file and just fall back to the default.